pkg/provider/azure_instances.go

```diff
@@ -304,6 +340,15 @@ func (az *Cloud) InstanceID(ctx context.Context, name types.NodeName) (string, e
 	return az.VMSet.GetInstanceIDByNodeName(nodeName)
 }
 
+func (az *Cloud) getNodeProviderIDByNodeName(ctx context.Context, name types.NodeName) (string, error) {
+	providerID, err := cloudprovider.GetInstanceProviderID(ctx, az, name)
```

Comment:
use `az.InstanceID()` instead?

Reply:
The `cloudprovider.GetInstanceProviderID` put a prefix `azure://` onto the `az.InstanceID`, which is the standard provider ID format. Some regexps need this such as https://github.com/kubernetes-sigs/cloud-provider-azure/blob/d12fc98b424d6a33e546679577825daccbc45165/pkg/provider/azure_vmss.go#L54

Reply:
OK, then it makes sense here.